### ISSUE TITLE
Run-time shape mismatch for XRTExecute argument

### ISSUE THREAD

**Issue by gakshaygupta** [pytorch/xla#1791]
## ❓ Questions and Help
Hi, I am a beginner. I have implemented a CNN model to be trained on google cloud TPU. The problem with the implementation is that after 130 steps I get an error and I am unable to understand what the error is implying. I am attaching the text file containing the output of the console. If required I will attach my code also.
Thanks in advance.
[error.txt](https://github.com/pytorch/xla/files/4361664/error.txt)

**Comment by gakshaygupta** [comment]
I have narrowed the error down. The error is happening when the model is performing backwards pass. But still unable to understand what the error is implying.

**Comment by dlibenzi** [comment]
Are you running on GCE or Colab?
Are you using nightly Docker/VM-images and pytorch-nightly TPU software?

**Comment by gakshaygupta** [comment]
I am using GCE on  PyTorch-nightly TPU software.

**Comment by dlibenzi** [comment]
Is your VM also up-to-date with nightly software?

**Comment by gakshaygupta** [comment]
yes

**Comment by gakshaygupta** [comment]
I have tried decreasing the batch size from 24 to 8 and still, the error happens at the 130th step. Also, I changed the seed for this run. I think the problem is with the optimization algorithm I am currently using adam with lr=0.0002 .

**Comment by dlibenzi** [comment]
Anything "special" happening at step 130?
Does it happen with SGD?

**Comment by gakshaygupta** [comment]
I tried with SGD for 1000 steps and error didn't happen. And nothing special is happening at step 130.

**Comment by dlibenzi** [comment]
Any chance you can attach a fully working model code?

**Comment by gakshaygupta** [comment]
https://github.com/gakshaygupta/Stereo-disparity

**Comment by gakshaygupta** [comment]
The train.py contains the code for dataloader and training. down_convolution.p, up_convolution and dispnet.py contain the code for the model.

**Comment by gakshaygupta** [comment]
to run the code use main.py

**Comment by dlibenzi** [comment]
Yes, but you need to give the parameters you are running it with.

**Comment by gakshaygupta** [comment]
sorry I forgot to provide that. Here they are batch_size=24  iterations=1000  log_interval=10 num_workers=2.

**Comment by dlibenzi** [comment]
Any chance you can give it a spin with the recent nightly?
We have fixed a bug yesterday which might be related to this.

**Comment by gakshaygupta** [comment]
okay

**Comment by gakshaygupta** [comment]
> Any chance you can give it a spin with the recent nightly?
> We have fixed a bug yesterday which might be related to this.
I ran with adam for 1000 iterations and error didn't happen this time but while performing optimizer step the time taken is more compared to that of SGD. The time taken for adam is around 30 x to 40x compared to that of SGD but still in manageable limits.

**Comment by dlibenzi** [comment]
ADAM is a lot more expensive than SGD, and extra cost cost is proportional to the number of parameters.
If you can put your code in a Colab reproducible notebook (or just post the debug run of 10..15 steps) we could take a look.
Colab debug template like:

https://github.com/pytorch/xla/blob/master/contrib/colab/issue-report.ipynb
